napari/layers/utils/style_encoding.py

```diff
@@ -100,7 +100,15 @@ def _json_encode(self) -> dict:
         """
 
 
-class _ConstantStyleEncoding(EventedModel, Generic[StyleValue, StyleArray]):
+class _StyleEncodingModel(EventedModel):
+    class Config:
+        # Match fields exactly so that dicts can be sensibly parsed.
+        extra = 'forbid'
```

Comment:
I honestly have no idea what this is doing? Probably why you wanted @tlambert03 to review, but possibly a more extnesive comment and class docstring would be helpful for future code archeologists?

Reply:
Good call. At a minimum, a link to the pydantic docs on the config would probably be a good idea. But I can also expand the comment.

We might actually want to put this config value in `EventedModel` - it forbids extra initialization parameters, whereas by default pydantic ignores them. This is particularly useful for color encodings because the different types have some overlapping field names.

Reply:
Done. @tlambert03 : not sure if this config value should go in `EventedModel.Config`. I prefer it to the default of `'ignore'`, but it could break current usage elsewhere.

Reply:
yeah, i'd be ok with it being the default as well (but, wouldn't be surprised if things broke if it got changed)